general-superstaq/general_superstaq/_models.py

```diff
@@ -0,0 +1,291 @@
+    max_priority: int | None = pydantic.Field(None)
+    submitted_before: datetime.datetime | None = pydantic.Field(None)
+    submitted_after: datetime.datetime | None = pydantic.Field(None)
+
+
+class UserTokenResponse(DefaultPydanticModel):
+    """Model for returning a user token to the client, either when adding a new user or
+    regenerating the token."""
+
+    email: pydantic.EmailStr
+    token: str
+
+
+class BalanceResponse(DefaultPydanticModel):
+    """Model for returning a single user balance."""
+
+    email: pydantic.EmailStr
+    balance: float
+
+
+class UserInfo(DefaultPydanticModel):
+    """Model for the user info returned to the client."""
+
+    name: str
+    """User name."""
+    email: pydantic.EmailStr
+    """User email."""
+    role: str
+    """User role."""
+    balance: float
+    """User balance"""
+    token: str
+    """User API token."""
+    user_id: uuid.UUID
+    """User id"""
+
+
+class UserQuery(DefaultPydanticModel):
+    """Model for querying the database to retrieve users. Use of lists implied logical OR. Providing
+    multiple fields (e.g. name and email) implies logical AND."""
+
+    name: list[str] | None = pydantic.Field(None)
+    """List of user names to filter for."""
+    email: list[pydantic.EmailStr] | None = pydantic.Field(None)
+    """List of user emails to filter for."""
+    role: list[str] | None = pydantic.Field(None)
+    """List of user roles to filter for."""
+    user_id: list[uuid.UUID] | None = pydantic.Field(None)
+    """List of explicit user IDs to filter for."""
+
+
+class AQTConfigs(DefaultPydanticModel):
+    """Model for AQT configs."""
+
+    pulses: str
+    """The serialized pulses."""
+    variables: str
+    """The serialized variables."""
+
+
+class NewUser(DefaultPydanticModel):
+    """Model for creating new users."""
+
+    name: str
+    """User name."""
+    email: pydantic.EmailStr
+    """User email."""
+    role: str | None = pydantic.Field(None)
+    """User role."""
+    initial_balance: float | None = pydantic.Field(None)
+    """Initial balance."""
+
+
+class UpdateUserDetails(DefaultPydanticModel):
+    """Model for requests which modify user details"""
+
+    name: str | None = pydantic.Field(None)
+    """New user name."""
+    role: str | None = pydantic.Field(None)
+    """New user role."""
+    balance: float | None = pydantic.Field(None)
+    """New user balance"""
+
+
+class TargetInfoModel(DefaultPydanticModel):
+    """Model for the info of a target."""
+
+    target_name: str
+    """The target name"""
+    supports_submit: bool
+    """Targets allow job submission."""
+    supports_submit_qubo: bool
+    """Targets allows QUBO submission."""
+    supports_compile: bool
+    """Target allows circuit compilation."""
+    available: bool
+    """Target is currently available."""
+    retired: bool
+    """Target is retired."""
+    simulator: bool
+    """Target is simulator"""
+
+
+class GetTargetsFilterModel(DefaultPydanticModel):
```

Comment:
could this and the above model be combined?

Reply:
Although the field names are the same they represent different things so I feel they should remain as separate models. Also on server side some logic is added to the to the FilterModel